jax/extend/backend.py

```diff
@@ -30,5 +30,18 @@
 from jax._src import (
     util as _util
 )
-add_clear_backends_callback = _util.cache_clearing_funs.add  # type: ignore
-del _util
+
+class _BackendCacheDeprecated:
+  def __init__(self, cache_clear: Callable):
+    self._cache_clear = cache_clear
+
+  def cache_clear(self):
+    self._cache_clear()
+
+
+def add_clear_backends_callback(cache_clear: Callable):
+  # DEPRECATED: use `util.register_cache`.
```

Comment:
Let's just make this change internally. It's only used in 1 place. No need to do this workaround.

Reply:
Technically this is part of `jax.extend`, which is a public API. It is true it was undocumented, but I prefer to make this change more carefully. I have also added a CHANGELOG entry.

Reply:
No, it's not public. THere's no need for this and no need for changelog entry too.

Since it is jax.extend and 0.7 is a breaking release, please do all the changes in a single CL. I don't want more tech debt related to caches especially this callback thing.